packages/account/src/providers/provider.ts

```diff
@@ -906,6 +923,54 @@ Supported fuel-core version: ${supportedVersion}.`
     return new TransactionResponse(transactionRequest, this, abis, subscription);
   }
 
+  /**
+   * Submits a transaction to the chain and awaits it's status response.
+   *
+   * @param transactionRequestLike - the request to submit.
+   * @param sendTransactionParams - The provider send transaction parameters (optional).
+   * @returns A promise that resolves to a settled transaction.
+   */
+  async sendTransactionAndAwaitStatus(
+    transactionRequestLike: TransactionRequestLike,
+    { estimateTxDependencies = true }: ProviderSendTxParams = {}
+  ): Promise<SendAndAwaitStatusResponse> {
+    const transactionRequest = transactionRequestify(transactionRequestLike);
+    if (estimateTxDependencies) {
+      await this.estimateTxDependencies(transactionRequest);
+    }
+
+    this.validateTransaction(transactionRequest);
+
+    const encodedTransaction = hexlify(transactionRequest.toTransactionBytes());
+
+    const transactionId = transactionRequest.getTransactionId(this.getChainId());
+    this.#cacheInputs(transactionRequest.inputs, transactionId);
+
+    const subscription = (await this.operations.submitAndAwaitStatus({
+      encodedTransaction,
+    })) as AsyncIterable<GqlSubmitAndAwaitStatusSubscription>;
+
+    for await (const sub of subscription) {
+      const statusChange = sub.submitAndAwaitStatus;
+      if (statusChange.type === 'SqueezedOutStatus') {
+        this.#uncacheInputs(transactionId);
+        throw new FuelError(
+          ErrorCode.TRANSACTION_SQUEEZED_OUT,
+          `Transaction Squeezed Out with reason: ${statusChange.reason}`
+        );
+      }
+      if (statusChange.type !== 'SubmittedStatus') {
+        return {
+          transactionId,
+          status: statusChange.type,
+          receipts: statusChange.receipts.map(processGqlReceipt),
+        };
+      }
+    }
+
+    return { transactionId, status: 'unknown', receipts: [] };
+  }
+
```

Comment:
```suggestion
  async sendTransactionAndAwaitStatus(
    transactionRequestLike: TransactionRequestLike,
    providerSendTxParams: ProviderSendTxParams = {}
  ): Promise<SendAndAwaitStatusResponse> {
    const transactionResponse = await this.sendTransaction(transactionRequestLike, providerSendTxParams);
    
    const result = await transactionResponse.waitForResult();
    
    return {
      transactionId: result.id,
      status: result.status,
      receipts: result.receipts,
    };
  }
```

It seems to me that we could do it as suggested above and wouldn't lose anything while not duplicating the tx preparation and subscription parsing logic and having to maintain it in both places. Am I missing something? By doing what I mentioned in https://github.com/FuelLabs/fuels-ts/issues/3536#issuecomment-2569237898 it wouldn't incur any additional overhead with regards to network calls.

Reply:
I've discussed it a bit further [here](https://github.com/FuelLabs/fuels-ts/issues/3545), but also I think there is room to provide something less abstract that just submits and awaits status without any other lifting provided by the SDK (that in `waitForResult`. Some fixes may come out of #3545, but I think there is still a need for this. This has been requested by SDK users.

Reply:
I will be implementing this. There is an improvement that can be made to `waitForResult` that will be done in #3536 that will mitigate any unnecessary calls for gas price, and as we are passing down the open subscription then this is performing the same function as my code, but DRY. 

There is additional work that will be done in #3545 for when the subscription is not passed to `TransactionResponse` via connector, but that will be dealt with there. This feature helps mitigate that by allowing a connector user package the transaction, then send it via `sendAndAwaitStatus` to maintain the subscription.

Reply:
Now out of date, updates made